Code/Framework/AzCore/AzCore/std/allocator_static.h

```diff
@@ -126,21 +107,13 @@ namespace AZStd
     template<AZStd::size_t Size, AZStd::size_t Alignment>
     AZ_FORCE_INLINE bool operator==(const static_buffer_allocator<Size, Alignment>& a, const static_buffer_allocator<Size, Alignment>& b)
     {
-        if (&a == &b)
-        {
-            return true;
-        }
-        return false;
+        return static_cast<bool>(&a == &b);
```

Comment:
Why the static cast.
This just comparing two pointers.

Reply:
Updated to remove the static cast.